quinn-proto/src/config.rs

```diff
@@ -838,6 +846,54 @@ impl ServerConfig {
         self.preferred_address_v6 = address;
         self
     }
+
+    /// Maximum number of [`Incoming`][crate::Incoming] to allow to exist at a time
+    ///
+    /// An [`Incoming`][crate::Incoming] comes into existence when an incoming connection attempt
+    /// is received and stops existing when the application either accepts it or otherwise disposes
+    /// of it. While this limit is reached, new incoming connection attempts are immediately
+    /// refused. Larger values have greater worst-case memory consumption, but accommodate greater
+    /// application latency in handling incoming connection attempts.
+    ///
+    /// The default value is set to 65536. Assuming each [`Incoming`][crate::Incoming] accounts for
+    /// little over 1200 bytes of memory at most, this should limit memory consumption from this to
+    /// under 100 MiB--a generous amount that still prevents memory exhaustion in most contexts.
+    pub fn max_incoming(&mut self, max_incoming: usize) -> &mut Self {
+        self.max_incoming = max_incoming;
+        self
+    }
+
+    /// Maximum number of received bytes to buffer for each [`Incoming`][crate::Incoming]
+    ///
+    /// An [`Incoming`][crate::Incoming] comes into existence when an incoming connection attempt
+    /// is received and stops existing when the application either accepts it or otherwise disposes
+    /// of it. This limit governs only packets received within that period, and does not include
+    /// the first packet. Packets received in excess of this limit are dropped, which may cause
+    /// 0-RTT data to have to be retransmitted.
```

Comment:
"0-RTT or handshake data", or maybe just "may temporarily delay the connection".

Reply:
I like specifying the specific types of packets because it seems noteworthy that in some cases (specific patterns of handshakes and no early data) it might be fine to set this to 0.

Reply:
Done.